internal/datastore/postgres/postgres.go

```diff
@@ -134,22 +134,25 @@ func newPostgresDatastore(
 		return nil, fmt.Errorf(errUnableToInstantiate, err)
 	}
 
-	// Setup the default custom plan setting, if applicable.
-	pgConfig, err := defaultCustomPlan(parsedConfig)
-	if err != nil {
-		return nil, fmt.Errorf(errUnableToInstantiate, err)
-	}
+	// By default, pgx uses prepared statements for all queries. This disables
+	// the use of prepared statements in order to work around a bug impacting
+	// performance documented here:
+	// https://spicedb.dev/d/force-custom-plan.
+	// The docs for the different execution modes offered by pgx may be found
+	// here:
+	// https://pkg.go.dev/github.com/jackc/pgx/v5#QueryExecMode
+	parsedConfig.ConnConfig.DefaultQueryExecMode = pgx.QueryExecModeCacheDescribe
```

Comment:
The query exec mode can also be provided via the DB DSN. By hardcoding this here we are effectively overriding any value provided. While I don't think there is a reason for users to change this, it would be helpful in case this causes a regression to any deployment out there and folks need a way to revert back to the original behaviour.

Reply:
This appears to be set with `default_query_exec_mode=cache_describe`. I'm not sure how you knew this, but I only see examples in `pgx` code/tests. Thanks for catching! Agreed that having an escape hatch is super valuable. I'll try to get that in place.